android/src/main/java/org/mozilla/experiments/nimbus/FakeGeneratedFeatureConfig.kt

```diff
@@ -0,0 +1,79 @@
+package org.mozilla.experiments.nimbus
+
+import android.graphics.Color
+
+// FAKE GENERATED CODE (not yet)
+// This shows two different models of accessing feature properties in a type safe manner.
+// They differ in when they do the type coercion.
+// However, they also show us a way for not passing JSON over the FFI, but still have a nice API.
+
+// This example does all its work in a secondary constructor, then discards to the `variables` object.
+// This has this has the benefit of being easy to use in a testing environment.
+// If we know that every key is going to be used, this is the version to use.
+// Because it has multiple constructors, this one (the data-class) will be better for re-use.
+data class ButtonFeatureConfig1(
+    val buttonColor: Int,
+    val buttonText: String
+) {
+    constructor(variables: Variables) : this(
+        variables.getColor("buttonColor") ?: Color.BLUE,
+        variables.getText("buttonText") ?: "Okay then"
+    )
+}
+
+// This example does its work bit by bit, lazily grabbing the values out of JSON as it goes.
+// If we think that not every key is going to be used, then this will be the less CPU intensive.
+class ButtonFeatureConfig2(
+    private val variables: Variables
+) {
+    val buttonColor: Int by lazy { variables.getColor("buttonColor") ?: Color.BLUE }
```

Comment:
Ah perfect. I was going to bring this up as something that's pretty useful if the feature starts getting kind of big (i'm running into this on Desktop right now)